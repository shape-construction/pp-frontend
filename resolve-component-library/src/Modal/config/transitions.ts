import type { TransitionClasses } from '@headlessui/react';

export type Transition =
  | 'bottom-to-top'
  | 'top-to-bottom'
  | 'left-to-right'
  | 'right-to-left'
  | 'fade'
  | 'none';

export type Configurations = Record<Transition, TransitionClasses>;

export const transitions: Configurations = {
  'fade': {
    enter: 'ease-out duration-150',
    enterFrom: 'opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95',
    enterTo: 'opacity-100 translate-y-0 sm:scale-100',
    leave: 'ease-in duration-150',
    leaveFrom: 'opacity-100 translate-y-0 sm:scale-100',
    leaveTo: 'opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95',
  },
  'bottom-to-top': {
    enter: 'transition ease-in-out duration-150 transform',
    enterFrom: 'translate-y-full',
    enterTo: '-translate-y-auto',
    leave: 'transition ease-in-out duration-150 transform',
    leaveFrom: '-translate-y-auto',
    leaveTo: 'translate-y-full',
  },
  'top-to-bottom': {
    enter: 'transition ease-in-out duration-150 transform',
    enterFrom: '-translate-y-full',
    enterTo: 'translate-y-0',
    leave: 'transition ease-in-out duration-150 transform',
    leaveFrom: 'translate-y-0',
    leaveTo: '-translate-y-full',
  },
  'left-to-right': {
    enter: 'transition ease-in-out duration-150 transform',
    enterFrom: '-translate-x-full',
    enterTo: 'translate-x-0',
    leave: 'transition ease-in-out duration-150 transform',
    leaveFrom: 'translate-x-0',
    leaveTo: '-translate-x-full',
  },
  'right-to-left': {
    enter: 'transition ease-in-out duration-150 transform',
    enterFrom: 'translate-x-full',
    enterTo: '-translate-x-auto',
    leave: 'transition ease-in-out duration-150 transform',
    leaveFrom: '-translate-x-auto',
    leaveTo: 'translate-x-full',
  },
  'none': {},
};
