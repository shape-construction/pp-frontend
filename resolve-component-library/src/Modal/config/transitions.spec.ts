import { transitions } from './transitions';

describe('dialog transitions', () => {
  it.each(Object.entries(transitions))('matches %s configuration', (transition) => {
    expect(transitions[transition]).toMatchSnapshot();
  });
});
