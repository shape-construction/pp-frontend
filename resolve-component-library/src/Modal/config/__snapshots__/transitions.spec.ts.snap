// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog transitions matches bottom-to-top configuration 1`] = `
{
  "enter": "transition ease-in-out duration-150 transform",
  "enterFrom": "translate-y-full",
  "enterTo": "-translate-y-auto",
  "leave": "transition ease-in-out duration-150 transform",
  "leaveFrom": "-translate-y-auto",
  "leaveTo": "translate-y-full",
}
`;

exports[`dialog transitions matches fade configuration 1`] = `
{
  "enter": "ease-out duration-150",
  "enterFrom": "opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95",
  "enterTo": "opacity-100 translate-y-0 sm:scale-100",
  "leave": "ease-in duration-150",
  "leaveFrom": "opacity-100 translate-y-0 sm:scale-100",
  "leaveTo": "opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95",
}
`;

exports[`dialog transitions matches left-to-right configuration 1`] = `
{
  "enter": "transition ease-in-out duration-150 transform",
  "enterFrom": "-translate-x-full",
  "enterTo": "translate-x-0",
  "leave": "transition ease-in-out duration-150 transform",
  "leaveFrom": "translate-x-0",
  "leaveTo": "-translate-x-full",
}
`;

exports[`dialog transitions matches none configuration 1`] = `{}`;

exports[`dialog transitions matches right-to-left configuration 1`] = `
{
  "enter": "transition ease-in-out duration-150 transform",
  "enterFrom": "translate-x-full",
  "enterTo": "-translate-x-auto",
  "leave": "transition ease-in-out duration-150 transform",
  "leaveFrom": "-translate-x-auto",
  "leaveTo": "translate-x-full",
}
`;

exports[`dialog transitions matches top-to-bottom configuration 1`] = `
{
  "enter": "transition ease-in-out duration-150 transform",
  "enterFrom": "-translate-y-full",
  "enterTo": "translate-y-0",
  "leave": "transition ease-in-out duration-150 transform",
  "leaveFrom": "translate-y-0",
  "leaveTo": "-translate-y-full",
}
`;
